.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -15,7 +15,7 @@ Fixes / closes #????
 - [ ] Added / updated test-suite
 <!-- If this is a feature pull request / breaks compatibility: -->
 <!-- (Otherwise, remove these lines.) -->
-- [ ] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
+- [ ] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified, and including reflecting any changes to the grammar in `common.edit_mlg` and running `make doc_gram_rsts`).
```

Comment:
Not all the grammar changes will require a change to `common.edit_mlg` (fortunately!). So I propose to simplify the wording to:
```suggestion
- [ ] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified, and including reflecting any changes to the grammar by running `make doc_gram_rsts`).
```

Reply:
How are users supposed to know if such an edit is necessary?  I have a sight preference for mentioning editing the file, so that people who edit the grammar can at least tag the doc team if they think they might me missing something.

Reply:
(In my mind, the goal of this checklist is to prevent as much as possible cases where people forget to synchronize or update something without realizing they've forgotten.)

Reply:
>How are users supposed to know if such an edit is necessary?

If an edit is required, the `doc_gram_rsts` target should fail with an error message mentioning the file. I don't expect that there would be any case when the file should be edited but the target doesn't fail.

Reply:
This seems like magic; how does it work?  Does it even work with tactic notations that modify the grammar?

Reply:
The tool helps, but it doesn't do everything.  One should review the changes it makes.  The things it does:
- updated syntax in a `:tacn:` or `:cmd:` - does matching based on the tactic/command name.  This updating works well.
- adding or changing productions of an existing nonterminal shown in `.. insertprodn / .. prodn::` will appear automatically, but may need to be reordered for the sake of readability in `orderedGrammar`
- adding new nonterminals - almost always need to adjust `orderedGrammar` and/or `insertprodn` to show the nonterminal in a sensible place
- if an existing nonterminal is renamed, the ordering and text may need adjustment.  The tool is unable to identify renamings
- `:tacv:` and `:cmdv:` are not handled, but we're in the process of removing them

As we've been revising the doc, we've frequently edited productions to make them more readable.  Frequently we rename nonterminals (and later feed these changes back into the grammar), splice nonterminals (substitute the RHS in place of the LHS everywhere and drop the symbol).  We also edit parts of the grammar that are hard to understand--they may be written in a particular way so the parser can handle them or for developer convenience.

The tool only looks at `mlg` files.  It doesn't examine anything in `.v` files.  Of course, the set of available tactic notations depends on what files you have loaded.  Any Ltac tactics that we include in the doc are added in manually in `common.edit_mlg`.

Reply:
I'm sure it's possible to recover tactic notations from the parser (some assembly required).  But the doc currently doesn't cover notations, so not on my "to do" list.